cmd/startup/startup.go

```diff
@@ -0,0 +1,308 @@
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/api/validate"
+	"github.com/openshift/openshift-azure/pkg/arm"
+	"github.com/openshift/openshift-azure/pkg/cluster"
+	"github.com/openshift/openshift-azure/pkg/tls"
+	azureclientstorage "github.com/openshift/openshift-azure/pkg/util/azureclient/storage"
+	"github.com/openshift/openshift-azure/pkg/util/configblob"
+	"github.com/openshift/openshift-azure/pkg/util/log"
+	"github.com/openshift/openshift-azure/pkg/util/template"
+)
+
+var (
+	logLevel         = flag.String("loglevel", "Debug", "valid values are Debug, Info, Warning, Error")
+	runningUnderTest = flag.Bool("runningundertest", false, "TestConfig.RunningUnderTest")
+	dnsHostname      = flag.String("dnshostname", "", "hostname -d")
+	gitCommit        = "unknown"
+)
+
+const (
+	origin = "/etc/origin"
+	etcd   = "/etc/etcd"
+)
+
+type startup struct {
+	blob azureclientstorage.Blob
+	log  *logrus.Entry
+}
+
+func (s *startup) writeKubeconfigs(cs *api.OpenShiftManagedCluster) error {
+	b, err := yaml.Marshal(cs.Config.AdminKubeconfig)
+	if err != nil {
+		return err
+	}
+	err = ioutil.WriteFile(path.Join(origin, "node/node.kubeconfig"), b, 0600)
+	if err != nil {
+		return err
+	}
+	err = ioutil.WriteFile(path.Join(origin, "master/admin.kubeconfig"), b, 0600)
+	if err != nil {
+		return err
+	}
+	b, err = yaml.Marshal(cs.Config.MasterKubeconfig)
+	if err != nil {
+		return err
+	}
+	err = ioutil.WriteFile(path.Join(origin, "master/openshift-master.kubeconfig"), b, 0600)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *startup) writeTemplatedFiles(cs *api.OpenShiftManagedCluster) error {
+	tc := api.TestConfig{RunningUnderTest: *runningUnderTest}
+	b, err := ioutil.ReadFile("/etc/hostname")
+	if err != nil {
+		return err
+	}
+	hostname := strings.TrimSpace(string(b))
+	s.log.Infof("hostname is \"%s\"", hostname)
+	s.log.Infof("dnsHostname %s", *dnsHostname)
+
+	files := []string{
+		path.Join(etcd, "etcd.conf"),
+		path.Join(origin, "master/master-config.yaml"),
+		path.Join(origin, "master/session-secrets.yaml"),
+		path.Join(origin, "master/scheduler.json"),
+		path.Join(origin, "node/pods/etcd.yaml"),
+		path.Join(origin, "node/pods/apiserver.yaml"),
+		path.Join(origin, "node/pods/controllers.yaml"),
+	}
+	if hostname == "master-000000" {
+		files = append(files, path.Join(origin, "node/pods/sync.yaml"))
+	}
+
+	for _, destination := range files {
+		s.log.Debugf("processing template %s", destination)
+		templateFileName := path.Base(destination)
+		templateFile, err := arm.Asset(templateFileName)
+		if err != nil {
+			return err
+		}
+		b, err := template.Template(string(templateFile), nil, cs, map[string]interface{}{
+			"Hostname":    hostname,
+			"DNSHostname": *dnsHostname,
+			"TestConfig":  tc,
```

Comment:
use cs.Config.RunningUnderTest instead

Reply:
should be able to remove `"TestConfig": tc,`

Reply:
TestConfig is used